src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/replication.py

```diff
@@ -0,0 +1,122 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.util import CLIError
+
+from azure.mgmt.containerregistry.v2017_10_01.models import ReplicationUpdateParameters
+
+from ._factory import get_acr_service_client
+from ._utils import (
+    get_resource_group_name_by_registry_name,
+    premium_registry_validation
+)
+
+
+REPLICATIONS_NOT_SUPPORTED = 'Replications are only supported for managed registries in Premium SKU.'
+
+
+def acr_replication_list(registry_name, resource_group_name=None):
+    """Lists all the replications for the specified container registry.
+    :param str registry_name: The name of container registry
+    :param str resource_group_name: The name of resource group
+    """
+    _, resource_group_name = premium_registry_validation(
+        registry_name, resource_group_name, REPLICATIONS_NOT_SUPPORTED)
+    client = get_acr_service_client().replications
+
+    return client.list(resource_group_name, registry_name)
```

Comment:
Can this command be implemented **not** as a custom command? The `premium_registry_validation` can serve as a validator.

Reply:
I think it has to be a custom command because we have been not requiring resource group so we need the custom logic to find resource group. Can a validator be applied at command level (as opposed to parameter level)?